ui/src/components/header.js

```diff
@@ -1,23 +1,42 @@
 import { Link } from 'react-router-dom';
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useState } from 'react';
 import '../styles/header.css';
 
-const Header = ({ siteTitle }) => (
-  <header className="header-container">
-    <div className="header-title">
-      <Link to="/" className="header-title__text">
-        {siteTitle}
-      </Link>
-      <input type="text" placeholder="Look For Group..." className="search-field header-title__search" />
-    </div>
-    <div className="header-login">
-      <Link to="/login/" className="input-button header-login__button">
-        Login
-      </Link>
-    </div>
-  </header>
-);
+const Header = ({ siteTitle }) => {
+  const [searchTerm, setSearchTerm] = useState('');
+
+  return (
+    <header className="header-container">
+      <div className="header-title">
+        <Link to="/" className="header-title__text">
+          {siteTitle}
+        </Link>
+        <input
+          data-testid="header-search"
+          onChange={(e) => {
+            e.preventDefault();
+            setSearchTerm(e.target.value);
+          }}
+          onKeyUp={(e) => {
+            if (e.keyCode === 13) {
+              e.preventDefault();
+              window.location.assign(`${window.location.origin}/search?term=${searchTerm}&page=0`);
```

Comment:
Is this interchangeable with `history.push('url_here')`? IIRC window.location.assign causes a refresh, and history.push should just load the page w/o refresh. You would need to import the history from the history.js file.